### DISCUSSION TITLE
How to gather loss across GPUs?

### DISCUSSION THREAD

**Discussion by tnq177** [deepspeedai/DeepSpeed#2377] [Q&A] [answered]
I'm trying to calculate perplexity. To do this, I need to sum all the loss across GPUs. It seems to me the loss return from `engine(input)` is for each GPU as when I print it out:

```
print(f"Rank = {engine.global_rank} ,", nll_loss.item())
```

I got:
```
Rank = 1 , 1.7060546875
Rank = 2 , 1.169921875
Rank = 0 , 1.052734375
Rank = 6 , 0.240478515625
Rank = 4 , 1.2724609375
Rank = 7 , 0.81884765625
...
```

How do I gather the loss from all GPUs please?

**Comment by tjruwase** [comment, marked as answer]
You can do a [reduce](https://pytorch.org/docs/stable/distributed.html#torch.distributed.all_reduce) to sum the losses, such as [this](https://github.com/microsoft/DeepSpeedExamples/blob/e7c8cb767acddba8ad5d2c41fe18e30de7870d30/bing_bert/deepspeed_train.py#L108-L111).
